api/dao/hierarchy.py

```diff
@@ -375,7 +335,33 @@ def _update_hierarchy(container, container_type, metadata):
     project = metadata.get('project', {})
     if project.keys():
         project['modified'] = now
-        project_obj = _update_container({'_id': project_id}, {}, project, 'projects')
+        project_obj = _update_container_nulls({'_id': project_id}, project, 'projects')
+
+def _update_container_nulls(base_query, update, container_type):
+    coll_name = container_type if container_type.endswith('s') else container_type+'s'
+    cont = config.db[coll_name].find_one(base_query)
+    if cont is None:
+        raise APIStorageException('Failed to find {} object using the query: {}'.format(container_type, base_query))
+
+    bulk = config.db[coll_name].initialize_unordered_bulk_op()
+
+    if (update.get('metadata') is not None and
+       (cont.get('metadata') is None or cont.get('metadata',{}).keys == 0)):
```

Comment:
this condition could be replaced with 
```update.get('metadata') and not cont.get('metadata')```
as both `None` and `{}` are falsy in python.